cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -244,19 +244,19 @@ public void testLocalDatasetDeletion() throws Exception {
     ProgramId workflowId = new ProgramId(TEST_NAMESPACE2, WorkflowAppWithLocalDatasets.NAME, ProgramType.WORKFLOW,
                                          WorkflowAppWithLocalDatasets.WORKFLOW_NAME);
 
-    startProgram(workflowId.toId(), ImmutableMap.of("wait.file", waitFile.getAbsolutePath(),
-                                                    "done.file", doneFile.getAbsolutePath(),
-                                                    "dataset.*.keep.local", "true"));
+    startProgram(Id.Program.fromEntityId(workflowId), ImmutableMap.of("wait.file", waitFile.getAbsolutePath(),
```

Comment:
why not directly use `workflowId`?

Reply:
thats a more laborious change as the old `Id` classes are used at a lot of places in our code base and not all the internal API's have made additional methods to use the new `EntityId` parameters, then it will involve adding those new methods everywhere. However the goal for us is to not expose `Id` classes in any of the public modules (cdap-proto, cdap-client, cdap-api, cdap-cli). so its okay to have the old `Id` class in internal usages.